chain/db/pubsub.go

```diff
@@ -128,94 +261,120 @@ func (cs *ChainStore) GetFirstAvailableTopicBucket(topic string) int {
 	return -1
 }
 
-func (cs *ChainStore) GetTopicBucketsCount(topic string) uint32 {
+func (cs *ChainStore) GetFirstAvailableTopicBucket(topic string) int {
+	return cs.States.getFirstAvailableTopicBucket(topic)
+}
+
+func (sdb *StateDB) getTopicBucketsCount(topic string) uint32 {
 	lastBucket := uint32(0)
 
-	prefix := generateTopicKey(topic)
+	prefix := getTopicId(topic)
 
-	iter := cs.st.NewIterator(prefix)
+	iter := trie.NewIterator(sdb.trie.NodeIterator(prefix))
+	var lastKey []byte
 	for iter.Next() {
-		rk := bytes.NewReader(iter.Key())
+		lastKey = iter.Key
+	}
 
-		// read prefix
-		_, _ = serialization.ReadBytes(rk, uint64(len(prefix)))
+	rk := bytes.NewReader(lastKey)
 
-		bucket, _ := serialization.ReadUint32(rk)
+	// read prefix
+	_, _ = serialization.ReadBytes(rk, uint64(len(prefix)))
 
-		lastBucket = bucket
-	}
+	bucket, _ := serialization.ReadUint32(rk)
```

Comment:
1. Instead of putting logic about key here, what about creating a function that parse bucket from key? Same for other places 
2. We may be able to ignore write error as you mentioned, but read error should definitely be handled because the data may be corrupted or in different format
3. Can we guarantee that the last key is the key that has the largest bucket? I think it's safer to parse every key and save the largest bucket id

Reply:
1. There is no other places though? ALso
3. It should be ordered, having to parse every key can be costly performance-wise.

Reply:
Whether it's ordered depends on the serialization of Uint32. Specifically we use little endian, so it's not ordered. And even if we do, making this assumption here couples with the serialization encoding and is quite dangerous. On the other hand, parse this key should be very fast, and we only need to parse 1000 times at most

Reply:
We need to parse 1000*1000 times at most, because it goes over every subscriber of every bucket.

Reply:
3. done